modules/apps/web-experience/product-navigation/product-navigation-control-menu-api/src/main/java/com/liferay/product/navigation/control/menu/ProductNavigationControlMenuEntry.java

```diff
@@ -25,36 +25,157 @@
+	 */
 	public String getLabel(Locale locale);
 
+	/**
+	 * Returns the link CSS class which should be injected as the linkCssClass
+	 * attribute of <code>liferay-ui:icon</code> tag instance for the control
+	 * menu entry.
+	 *
+	 * @param  request the request with which the control menu entry is
+	 *         rendered
+	 * @return the linkCssClass attribute of <code>liferay-ui:icon</code> tag
+	 *         instance for the control menu entry
+	 */
 	public String getLinkCssClass(HttpServletRequest request);
 
+	/**
+	 * Returns the markup view string which should be injected as the markupView
+	 * attribute of <code>liferay-ui:icon</code> tag instance for the control
+	 * menu entry.
+	 *
+	 * @param  request the request with which the control menu entry is
+	 *         rendered
+	 * @return the markupView attribute of <code>liferay-ui:icon</code> tag
+	 *         instance for the control menu entry
+	 */
 	public String getMarkupView(HttpServletRequest request);
 
+	/**
+	 * Returns the URL which should be injected as the url attribute of
+	 * <code>liferay-ui:icon</code> tag instance for the control menu entry.
+	 *
+	 * @param  request the request with which the control menu entry is
+	 *         rendered
+	 * @return the url attribute of <code>liferay-ui:icon</code> tag instance
+	 *         for the control menu entry
+	 */
 	public String getURL(HttpServletRequest request);
 
+	/**
+	 * Renders the HTML that needs to be displayed when the control menu entry
+	 * body is displayed.
+	 *
+	 * @param  request the request with which the control menu entry body is
+	 *         rendered
+	 * @param  response the response with which the control menu entry body is
+	 *         rendered
+	 * @throws IOException if an IO exception occurs
+	 */
 	public boolean includeBody(
 			HttpServletRequest request, HttpServletResponse response)
 		throws IOException;
 
+	/**
+	 * Renders the HTML that needs to be displayed when the control menu entry
+	 * icon is displayed.
+	 *
+	 * @param  request the request with which the control menu entry icon is
+	 *         rendered
+	 * @param  response the response with which the control menu entry icon is
+	 *         rendered
+	 * @throws IOException if an IO exception occurs
+	 */
 	public boolean includeIcon(
 			HttpServletRequest request, HttpServletResponse response)
 		throws IOException;
 
+	/**
+	 * Returns <code>true</code> if the control menu entry should be
+	 * displayed in the context of specific request.
+	 *
+	 * @param  request the request with which the control menu entry is
+	 *         rendered
+	 * @return <code>true</code> if the control menu entry should be
+	 *         displayed in current request context; <code>false</code>
+	 *         otherwise
+	 */
 	public boolean isShow(HttpServletRequest request) throws PortalException;
 
+	/**
+	 * Returns <code>true</code> if the control menu entry should be opened
+	 * in a dialog window or in the current window.
+	 *
```

Comment:
@pavel-savinov does this method return `false` if the entry should be opened in the current window? Or is it `true` for both a dialog window **AND** the current window?